h2o-core/src/main/java/water/util/JSONValue.java

```diff
@@ -0,0 +1,66 @@
+package water.util;
+
+import water.Iced;
+import water.api.Schema;
+
+import java.lang.reflect.Array;
+import java.util.Objects;
+
+public class JSONValue<V> extends Iced {
```

Comment:
@michalkurka any concern if I add this new class to core?
AutoML is posting whole payload in POST body as JSON.
This class is intended to provide a safe encapsulation of some parts of the body that can be parsed dynamically later.

See the test and the use in `POJOUtils` below.

Reply:
I think this is a very clean solution, I don't see any issue having it in core.